.github/workflows/build-ffmpeg.yml

```diff
@@ -24,102 +65,155 @@ jobs:
 
-      - name: Setup Dependencies
+      - name: Setup Dependencies Unix
+        if: ${{ matrix.os_type != 'windows' }}
         run: |
-          sudo apt-get update -qq && sudo apt-get -y install \
-            autoconf \
-            automake \
-            build-essential \
-            cmake \
-            git-core \
-            libass-dev \
-            libfreetype6-dev \
-            libgnutls28-dev \
-            libmp3lame-dev \
-            libsdl2-dev \
-            libtool \
-            libva-dev \
-            libvdpau-dev \
-            libvorbis-dev \
-            libxcb1-dev \
-            libxcb-shm0-dev \
-            libxcb-xfixes0-dev \
-            meson \
-            ninja-build \
-            pkg-config \
-            texinfo \
-            wget \
-            yasm \
-            zlib1g-dev
-
-          # NASM
-          sudo apt-get install nasm
-
-          # libopus
-          sudo apt-get install libopus-dev
-
-          # libx264... build from source to enable static library
-          # sudo apt-get install libx264-dev
-
-          # libx265... build from source to enable static library
-          # sudo apt-get install libx265-dev libnuma-dev
+          if [[ ${{ matrix.os_type }} == "linux" ]]; then
+            sudo apt-get update -qq && sudo apt-get -y install \
+              autoconf \
+              automake \
+              build-essential \
+              cmake \
+              git-core \
+              libass-dev \
+              libfreetype6-dev \
+              libgnutls28-dev \
+              libmp3lame-dev \
+              libopus-dev \
+              libsdl2-dev \
+              libtool \
+              libva-dev \
+              libvdpau-dev \
+              libvorbis-dev \
+              libxcb1-dev \
+              libxcb-shm0-dev \
+              libxcb-xfixes0-dev \
+              meson \
+              nasm \
+              ninja-build \
+              pkg-config \
+              texinfo \
+              wget \
+              yasm \
+              zlib1g-dev
+          elif [[ ${{ matrix.os_type }} == "macos" ]]; then
+            brew install \
+              automake \
+              fdk-aac \
+              git \
+              lame \
+              libass \
+              libtool \
+              libvorbis \
+              libvpx \
+              nasm \
+              opus \
+              sdl \
+              shtool \
+              texi2html \
+              theora \
+              wget \
+              x264 \
+              x265 \
+              xvid
+          fi
+
+      - name: Setup Dependencies Windows
+        if: ${{ matrix.os_type == 'windows' }}
+        uses: msys2/setup-msys2@v2
+        with:
+          update: true
+          install: >-
+            diffutils
+            git
+            make
+            pkg-config
+            mingw-w64-x86_64-cmake
+            mingw-w64-x86_64-gcc
+            mingw-w64-x86_64-nasm
```

Comment:
Also needs:
```
mingw-w64-x86_64-amf-headers
mingw-w64-x86_64-ffnvcodec-headers
```
The mingw packages for `x264` and `x265` allow you to build, but the shared libraries provided not allow you to create a static library. They will need to be compiled separately.